### PR TITLE
`azurerm_kubernetes_cluster`: Add `ingress_application_gateway_identity` export for add-on `ingress_application_gateway`

### LINKED ISSUE #11564: `ingress_application_gateway` AKS addon should export managed identity properties
Hi all. 

Recently, new version of provider has been released that finally allows to install appgw ingress controller as AKS addon via terraform. Sadly, compared to `az aks enable-addons -n AKS_NAME -g RG_NAME -a ingress-appgw --appgw-id APPGW_ID` command, it does not configure access for ingress controller addon to access appgw objects. 

What does it mean? When addon is installed via terraform, managed identity for ingress appgw controller is created in the same resource group where cluster exists, but no role is assigned to it. As a result, ingress controller running in the cluster is not able to configure appgw and is useless. Also, its properties are not exported to terraform code, thus I am unable to retrieve identity ID and assign role by myself in terraform.
I can see only option how to make this addon work and that is by assigning role manually.

Suggestions:
- export `identity` properties of managed identity that was created (similar to `oms_agent_identity`) so it can be reference in terraform code and assigned some role
- make `user_assigned_identity_id` attribute as optional for `addon_profile.ingress_application_gateway` resource so it can be provided externally (probably more work)

Relevant code in `az cli`: https://github.com/Azure/azure-cli/blob/561240abe913d98d861453c29640501965614890/src/azure-cli/azure/cli/command_modules/acs/custom.py#L1737

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform (and AzureRM Provider) Version

terraform: `v0.14.9`
azurerm provider: `v2.57.0`

### Affected Resource(s)

* `azurerm_kubernetes_cluster`

Specifically its part:
```hcl
   addon_profile {
      ingress_application_gateway {
        enabled    = true
        gateway_id = azurerm_application_gateway.ingress_appgw.id
      }
    }
```
Related to: https://github.com/terraform-providers/terraform-provider-azurerm/pull/11376

### PR DESCRIPTION
**Pull request by aristosvo**
Fixes #11564

## Acceptance tests

- [x] New functionality tested and passing
  ```
  ❯ make acctests SERVICE='containers' TESTARGS='-run=TestAccKubernetesCluster_addonProfileIngressApplicationGateway_appGatewayId'
  ==> Checking that code complies with gofmt requirements...
  ==> Checking that Custom Timeouts are used...
  ==> Checking that acceptance test packages are used...
  TF_ACC=1 go test -v ./azurerm/internal/services/containers -run=TestAccKubernetesCluster_addonProfileIngressApplicationGateway_appGatewayId -timeout 180m -ldflags="-X=github.com/terraform-providers/terraform-provider-azurerm/version.ProviderVersion=acc"
  2021/05/07 18:15:43 [DEBUG] not using binary driver name, it's no longer needed
  2021/05/07 18:15:44 [DEBUG] not using binary driver name, it's no longer needed
  === RUN   TestAccKubernetesCluster_addonProfileIngressApplicationGateway_appGatewayId
  === PAUSE TestAccKubernetesCluster_addonProfileIngressApplicationGateway_appGatewayId
  === CONT  TestAccKubernetesCluster_addonProfileIngressApplicationGateway_appGatewayId
  --- PASS: TestAccKubernetesCluster_addonProfileIngressApplicationGateway_appGatewayId (1302.15s)
  PASS
  ok      github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/containers    1306.024s
  ```

  ```
  ❯ make acctests SERVICE='containers' TESTARGS='-run=TestAccDataSourceKubernetesCluster_addOnProfileIngressApplicationGatewayAppGateway'
  ==> Checking that code complies with gofmt requirements...
  ==> Checking that Custom Timeouts are used...
  ==> Checking that acceptance test packages are used...
  TF_ACC=1 go test -v ./azurerm/internal/services/containers -run=TestAccDataSourceKubernetesCluster_addOnProfileIngressApplicationGatewayAppGateway -timeout 180m -ldflags="-X=github.com/terraform-providers/terraform-provider-azurerm/version.ProviderVersion=acc"
  2021/05/07 20:59:29 [DEBUG] not using binary driver name, it's no longer needed
  2021/05/07 20:59:30 [DEBUG] not using binary driver name, it's no longer needed
  === RUN   TestAccDataSourceKubernetesCluster_addOnProfileIngressApplicationGatewayAppGateway
  === PAUSE TestAccDataSourceKubernetesCluster_addOnProfileIngressApplicationGatewayAppGateway
  === CONT  TestAccDataSourceKubernetesCluster_addOnProfileIngressApplicationGatewayAppGateway
  --- PASS: TestAccDataSourceKubernetesCluster_addOnProfileIngressApplicationGatewayAppGateway (1341.64s)
  PASS
  ok      github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/containers  1345.299s
  ```

### CODE REVIEW

**Review by mbfrahry — CHANGES_REQUESTED** [review]
Hey @aristosvo! This looks mostly good minus a couple things I called out below! Thanks for taking the time to get this PR out

File: azurerm/internal/services/containers/kubernetes_cluster_data_source.go [hunk 1]
```diff
@@ -883,12 +903,18 @@ func flattenKubernetesClusterDataSourceAddonProfiles(profile map[string]*contain
...
+			return err
+		}
+
 		output := map[string]interface{}{
 			"enabled":              enabled,
 			"gateway_id":           gatewayId,
 			"effective_gateway_id": effectiveGatewayId,
 			"subnet_cidr":          subnetCIDR,
 			"subnet_id":            subnetId,
+			"oms_agent_identity":   ingressApplicationGatewayIdentity,
```
[the remark this answers is not in the document]
Comment (aristosvo) [reply azurerm/internal/services/containers/kubernetes_cluster_data_source.go]: Thanks!

File: website/docs/d/kubernetes_cluster.html.markdown [hunk 2]
```diff
@@ -270,6 +272,12 @@ An `ingress_application_gateway` block supports the following:
 
 * `subnet_id` - The ID of the subnet on which to create an Application Gateway, which in turn will be integrated with the ingress controller of this Kubernetes Cluster. This attribute is only set when `subnet_id` is specified when configuring the `ingress_application_gateway` addon.
 
+* `client_id` - The Client ID of the user-defined Managed Identity used by the Application Gateway.
```
[the remark this answers is not in the document]
Comment (aristosvo) [reply website/docs/d/kubernetes_cluster.html.markdown:281]: Yeah, sorry. Copied it without really paying attention, thanks!

**Comment by aristosvo** [comment]
Thanks @mbfrahry! Sorry to bother you with correcting my Friday night code 😅

I think this should do!

**Review by mbfrahry — APPROVED** [review]
No worries at all and LGTM! Thanks for taking care of that!

**Pull request merged**